frontend/packages/kubevirt-plugin/integration-tests/tests/vm.non-admin.scenario.ts

```diff
@@ -0,0 +1,77 @@
+import { $, browser, ExpectedConditions as until } from 'protractor';
+import { appHost, testName } from '@console/internal-integration-tests/protractor.conf';
+import * as loginView from '@console/internal-integration-tests/views/login.view';
+import { withResource, removeLeakableResource } from '@console/shared/src/test-utils/utils';
+import { createProject } from './utils/utils';
+import { vmConfig, getProvisionConfigs } from './vm.wizard.configs';
+import { ProvisionConfigName } from './utils/constants/wizard';
+import { VirtualMachine } from './models/virtualMachine';
+import { VM_ACTION } from './utils/consts';
+
+const JASMINE_DEFAULT_TIMEOUT_INTERVAL = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+const JASMINE_EXTENDED_TIMEOUT_INTERVAL = 1000 * 60 * 3;
+const {
+  BRIDGE_HTPASSWD_IDP = 'test',
+  BRIDGE_HTPASSWD_USERNAME = 'test',
+  BRIDGE_HTPASSWD_PASSWORD = 'test',
+} = process.env;
+const configName = ProvisionConfigName.URL;
+const leakedResources = new Set<string>();
+const provisionConfigs = getProvisionConfigs(testName);
+
+const provisionConfig = provisionConfigs.get(configName);
+provisionConfig.networkResources = [];
+provisionConfig.storageResources = [];
+
+describe('Kubevirt non-admin VM Creation', async () => {
+  beforeAll(async () => {
+    await browser.get(appHost);
+    await browser.sleep(3000); // Wait long enough for the login redirect to complete
+  });
+  describe('Non-Admin Flow Test', async () => {
+    beforeAll(() => {
+      // Extend the default jasmine timeout interval just in case it takes a while for the htpasswd idp to be ready
+      jasmine.DEFAULT_TIMEOUT_INTERVAL = JASMINE_EXTENDED_TIMEOUT_INTERVAL;
+    });
+    afterAll(() => {
+      // Set jasmine timeout interval back to the original value after these tests are done
+      jasmine.DEFAULT_TIMEOUT_INTERVAL = JASMINE_DEFAULT_TIMEOUT_INTERVAL;
+    });
+
+    it('Non-Admin Create-VM and Delete Flow Test', async () => {
+      const testNonAdminNamespace = `${testName}-non-admin`;
+      await loginView.login(
+        BRIDGE_HTPASSWD_IDP,
+        BRIDGE_HTPASSWD_USERNAME,
+        BRIDGE_HTPASSWD_PASSWORD,
+      );
+      expect(browser.getCurrentUrl()).toContain(appHost);
+      expect(loginView.userDropdown.getText()).toContain(BRIDGE_HTPASSWD_USERNAME);
+
+      // Navigate to Virtual Machines page
+      await browser.get(`${appHost}/k8s/ns/${testNonAdminNamespace}/virtualmachines`);
+
+      // Check to make sure Access is Restricted.
+      await browser.wait(
+        until.textToBePresentInElement($('.cos-status-box__title'), 'Restricted Access'),
+      );
+
+      await createProject(testNonAdminNamespace);
```

Comment:
I would say that all until and including the project creation can be done in beforeAll block.
Also, I think it would be better to only use one `describe` block.

Reply:
i'm fine with moving to 1 describe block.
about the project, I want to make sure the access is restricted to the namespace, I guess I can use `default` namespace instead?

Reply:
Yes `default` shoud be ok